code-studio/src/js/components/Grid.jsx

```diff
@@ -68,7 +76,76 @@ var Grid = React.createClass({
     selectedRow: React.PropTypes.number,
     selectedCol: React.PropTypes.number,
     skin: React.PropTypes.string.isRequired,
-    onSelectionChange: React.PropTypes.func.isRequired
+    onSelectionChange: React.PropTypes.func.isRequired,
+    setCopiedCells: React.PropTypes.func.isRequired,
+  },
+
+  /**
+   * When drag begins, record that we are now dragging and where we
+   * started from.
+   */
+  beginDrag: function (row, col) {
+    this.setState({
+      dragging: true,
+      dragStart: {row: row, col: col},
+    });
+  },
+
+  /**
+   * As the mouse moves over the cells, if we are dragging then record
+   * the latest cell we've moved over so we can highlight all selected
+   * cells appropriately.
+   */
+  moveDrag: function (row, col) {
+    if (this.state && this.state.dragging) {
+      this.setState({
+        dragCurrent: {row: row, col: col},
+      });
+    }
+  },
+
+  /**
+   * Once the drag ends, create a subarray of all selected cells and
+   * save it to our parent.
+   */
+  endDrag: function (row, col) {
+    var from = this.state.dragStart;
+    this.setState({
+      dragging: false,
+      dragStart: null,
+      dragCurrent: null
+    });
+
+    if (from.row == row && from.col == col) {
+      return;
+    }
+
+    var top = Math.min(from.row, row),
+        left = Math.min(from.col, col),
+        bottom = Math.max(from.row, row),
+        right = Math.max(from.col, col);
+
+    var cells = this.props.cells.slice(top, bottom + 1).map(function (row) {
+      return row.slice(left, right + 1).map(function (cell) {
+        return cell.serialize();
+      });
+    });
+
+    this.props.setCopiedCells(cells);
```

Comment:
I was going to say that 'copy' is not necessarily part of this logic and `onSelectionChange` would be a better callback function... until I realized that you already have `onSelectionChange` for the click-select behavior.  Should this be `onDragSelectionChange`?

Maybe better to leave as-is for now, and later unify the concept of 'current selection' so that click and drag both affect selection state (as in a spreadsheet) - but do that later, because I'm sure it requires solving all sorts of other questions about what to do when the selection is more than one cell.